File: packages/app/src/services/software/definitions/variants/discordCanary.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'discord-canary';
	public name = 'Discord Canary';
	public category = [SoftwareCategory.Social];
	public downloadName = 'DiscordCanarySetup.exe';
	public icon = 'discord-canary.png';
	public homepage = 'https://canary.discord.com';
	public parent = 'discord';

	public async resolveDownloadUrl() {
		return ok('https://canary.discord.com/api/downloads/distributions/app/installers/latest?channel=canary&platform=win&arch=x64');
	}
}

File: src/lib/software/variants/discord_canary.py
from src.lib.software import BaseSoftware

class DiscordCanary(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'DiscordCanarySetup.exe'
        self.silent_install_args = ['--silent']

    def resolve_download_url(self):
        self.url_resolved.emit('https://canary.discord.com/api/downloads/distributions/app/installers/latest?channel=canary&platform=win&arch=x64')
